src/EditorFeatures/CSharpTest/CodeActions/ConvertIfToSwitch/ConvertIfToSwitchFixAllTests.cs

```diff
@@ -0,0 +1,575 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.ConvertIfToSwitch;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions.ConvertIfToSwitch
+{
+    public class ConvertIfToSwitchFixAllTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new CSharpConvertIfToSwitchCodeRefactoringProvider();
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsFixAllOccurrences)]
+        public async Task ConvertIfToSwitchStatement_FixAllInDocument()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    int M(int i)
+    {
+        {|FixAllInDocument:|}if (i == 3)
+        {
+            return 0;
+        }
+        else
+        {
+            if (i == 6) return 1;
+            if (i == 7) return 1;
+            return 0;
+        }
+    }
+
+    int M2(int i)
+    {
+        if (i == 3)
+        {
+            return 0;
+        }
+        else
+        {
+            if (i == 6) return 1;
+            if (i == 7) return 1;
+            return 0;
+        }
+    }
+}",
+@"class C
+{
+    int M(int i)
+    {
+        switch (i)
+        {
+            case 3:
+                return 0;
+            case 6:
+                return 1;
+            case 7:
+                return 1;
+            default:
+                return 0;
+        }
+    }
+
+    int M2(int i)
+    {
+        switch (i)
+        {
+            case 3:
+                return 0;
+            case 6:
+                return 1;
+            case 7:
+                return 1;
+            default:
+                return 0;
+        }
+    }
+}");
+        }
```

Comment:
if you don't have one, please have test where one 'if' is nested in the other, and both become a distinct switch.  Note: i don't care what behavior this produces, as long as it doesn't crash.

Reply:
> if you don't have one, please have test where one 'if' is nested in the other, and both become a distinct switch

Sure, will add a test

Reply:
thanks!

Reply:
Added with https://github.com/dotnet/roslyn/pull/61874/commits/50e20479d96d4b922f7d102abdf37e8fe93da8dc